// @looker/sdk-rtl/lib/esm/authToken.js
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
export class AuthToken {
  constructor(token) {
    _defineProperty(this, "lagTime", 10);
    _defineProperty(this, "access_token", '');
    _defineProperty(this, "token_type", '');
    _defineProperty(this, "refresh_token", void 0);
    _defineProperty(this, "expires_in", 0);
    _defineProperty(this, "expiresAt", new Date());
    if (!token) token = {};
    this.setToken(token);
  }
  isActive() {
    if (!this.access_token) return false;
    if (!this.expiresAt) return false;
    var now = new Date();
    return this.expiresAt > now;
  }
  setToken(token) {
    this.access_token = token.access_token || '';
    this.token_type = token.token_type || '';
    this.expires_in = token.expires_in || 0;
    if (token.refresh_token) {
      this.refresh_token = token.refresh_token;
    }
    var exp = new Date();
    if (this.access_token && this.expires_in) {
      exp.setSeconds(exp.getSeconds() + this.expires_in - this.lagTime);
    } else {
      exp.setSeconds(exp.getSeconds() - this.lagTime);
    }
    this.expiresAt = exp;
    return this;
  }
  reset() {
    this.access_token = '';
    this.expires_in = 0;
  }
}
