// @looker/sdk/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  ILooker40SDK: true,
  Looker40SDK: true,
  Looker40SDKStream: true,
  functionalSdk40: true,
  sdkVersion: true,
  environmentPrefix: true,
  LookerExtensionSDK: true,
  BrowserSettings: true,
  LookerBrowserSDK: true
};
Object.defineProperty(exports, "BrowserSettings", {
  enumerable: true,
  get: function get() {
    return _browserSdk.BrowserSettings;
  }
});
Object.defineProperty(exports, "ILooker40SDK", {
  enumerable: true,
  get: function get() {
    return _methodsInterface.ILooker40SDK;
  }
});
Object.defineProperty(exports, "Looker40SDK", {
  enumerable: true,
  get: function get() {
    return _methods.Looker40SDK;
  }
});
Object.defineProperty(exports, "Looker40SDKStream", {
  enumerable: true,
  get: function get() {
    return _streams.Looker40SDKStream;
  }
});
Object.defineProperty(exports, "LookerBrowserSDK", {
  enumerable: true,
  get: function get() {
    return _browserSdk.LookerBrowserSDK;
  }
});
Object.defineProperty(exports, "LookerExtensionSDK", {
  enumerable: true,
  get: function get() {
    return _extensionSdk.LookerExtensionSDK;
  }
});
Object.defineProperty(exports, "environmentPrefix", {
  enumerable: true,
  get: function get() {
    return _constants.environmentPrefix;
  }
});
Object.defineProperty(exports, "functionalSdk40", {
  enumerable: true,
  get: function get() {
    return _funcs.functionalSdk40;
  }
});
Object.defineProperty(exports, "sdkVersion", {
  enumerable: true,
  get: function get() {
    return _constants.sdkVersion;
  }
});
var _methodsInterface = require("./4.0/methodsInterface");
var _methods = require("./4.0/methods");
var _streams = require("./4.0/streams");
var _funcs = require("./4.0/funcs");
Object.keys(_funcs).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _funcs[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _funcs[key];
    }
  });
});
var _models = require("./4.0/models");
Object.keys(_models).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _models[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _models[key];
    }
  });
});
var _constants = require("./constants");
var _extensionSdk = require("./extensionSdk");
var _browserSdk = require("./browserSdk");
