// @looker/design-tokens/dist/esm/color/fallbacks.js
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return typeof key === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (typeof input !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (typeof res !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
import { charcoal000, charcoal100, charcoal200, charcoal300, charcoal400, charcoal500, charcoal600, charcoal700, charcoal800, green000, green100, green300, green400, green500, purple000, purple100, purple300, purple400, red000, red100, red300, red400, red500, white } from '../legacy/palette';
import { defaultSpecifiableColors } from './defaultSpecifiableColors';
import { generateInteractive, generatePressed } from './utils';
import { generateExtendedStatefulColors } from './utils/generateStatefulColors';
export const fallbackBlends = {
  ui1: charcoal100,
  ui2: charcoal200,
  ui3: charcoal300,
  ui4: charcoal400,
  ui5: charcoal800,
  text1: charcoal400,
  text2: charcoal500,
  text3: charcoal600,
  text4: charcoal700,
  text5: charcoal800
};
export const fallbackStateful = _objectSpread({
  keySubtle: purple000,
  keyAccent: purple100,
  keyFocus: purple300,
  keyInteractive: generateInteractive(purple400),
  keyPressed: generatePressed(purple400),
  keyText: white,
  keyBorder: purple400,
  criticalSubtle: red000,
  criticalAccent: red100,
  criticalFocus: red300,
  criticalInteractive: generateInteractive(red500),
  criticalPressed: generatePressed(red500),
  criticalText: white,
  criticalBorder: red400,
  neutralSubtle: charcoal000,
  neutralAccent: charcoal100,
  neutralFocus: charcoal300,
  neutralInteractive: generateInteractive(charcoal500),
  neutralPressed: generatePressed(charcoal500),
  neutralText: white,
  neutralBorder: charcoal400,
  positiveSubtle: green000,
  positiveAccent: green100,
  positiveFocus: green300,
  positiveInteractive: generateInteractive(green500),
  positivePressed: generatePressed(green500),
  positiveText: white,
  positiveBorder: green400
}, generateExtendedStatefulColors(defaultSpecifiableColors));
