// @looker/sdk-rtl/lib/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.unquote = exports.matchModeString = exports.matchModeBinary = exports.matchCharsetUtf8 = exports.matchCharset = exports.isTrue = exports.isFalse = exports.boolDefault = void 0;
var matchCharset = exports.matchCharset = ';.*charset=';
var matchCharsetUtf8 = exports.matchCharsetUtf8 = "".concat(matchCharset, ".*\\butf-8\\b");
var matchModeString = exports.matchModeString = "(^application\\/.*(\\bjson\\b|\\bxml\\b|\\bsql\\b|\\bgraphql\\b|\\bjavascript\\b|\\bx-www-form-urlencoded\\b)|^text\\/|.*\\+xml\\b|".concat(matchCharset, ")");
var matchModeBinary = exports.matchModeBinary = '^image\\/|^audio\\/|^video\\/|^font\\/|^application\\/|^multipart\\/';
var isTrue = value => /^(true|t|yes|y|1)$/i.test(value);
exports.isTrue = isTrue;
var isFalse = value => /^(false|f|no|n|0)$/i.test(value);
exports.isFalse = isFalse;
var boolDefault = exports.boolDefault = function boolDefault(value) {
  var defaultBool = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;
  if (isTrue(value)) return true;
  if (isFalse(value)) return false;
  return defaultBool;
};
var unquote = value => {
  if (!value) return '';
  if (/^['"`]/.test(value)) {
    var quote = value.substring(0, 1);
    if (value.endsWith(quote)) return value.substring(1, value.length - 1);
  }
  return value;
};
exports.unquote = unquote;
