// @looker/design-tokens/dist/esm/color/utils/generateStatefulColors.js
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return typeof key === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (typeof input !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (typeof res !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
import darken from 'polished/lib/color/darken';
import lighten from 'polished/lib/color/lighten';
import { mixScaledColors } from './mixScaledColors';
export const generateInteractive = color => lighten(0.04, color);
export const generatePressed = color => darken(0.07, color);
export const accentBlendScale = 16;
export const generateStatefulColor = (background, color) => {
  return {
    subtle: mixScaledColors(10, color, background),
    accent: mixScaledColors(accentBlendScale, color, background),
    focus: mixScaledColors(60, color, background),
    interactive: generateInteractive(color),
    pressed: generatePressed(color),
    text: background,
    border: color
  };
};
export const generateExtendedStatefulColors = specifiable => {
  const {
    background,
    calculation,
    dimension,
    measure
  } = specifiable;
  const calculationColors = generateStatefulColor(background, calculation);
  const dimensionColors = generateStatefulColor(background, dimension);
  const measureColors = generateStatefulColor(background, measure);
  return {
    calculationSubtle: calculationColors.subtle,
    calculationAccent: calculationColors.accent,
    calculationFocus: calculationColors.focus,
    calculationInteractive: calculationColors.interactive,
    calculationPressed: calculationColors.pressed,
    calculationText: calculationColors.text,
    calculationBorder: calculationColors.border,
    dimensionSubtle: dimensionColors.subtle,
    dimensionAccent: dimensionColors.accent,
    dimensionFocus: dimensionColors.focus,
    dimensionInteractive: dimensionColors.interactive,
    dimensionPressed: dimensionColors.pressed,
    dimensionText: dimensionColors.text,
    dimensionBorder: dimensionColors.border,
    measureSubtle: measureColors.subtle,
    measureAccent: measureColors.accent,
    measureFocus: measureColors.focus,
    measureInteractive: measureColors.interactive,
    measurePressed: measureColors.pressed,
    measureText: measureColors.text,
    measureBorder: measureColors.border
  };
};
export const generateStatefulColors = (specifiable, derivatives) => {
  const {
    background,
    key,
    critical,
    positive
  } = specifiable;
  const {
    neutral
  } = derivatives;
  const extendedStatefulColors = generateExtendedStatefulColors(specifiable);
  const keyColors = generateStatefulColor(background, key);
  const criticalColors = generateStatefulColor(background, critical);
  const neutralColors = generateStatefulColor(background, neutral);
  const positiveColors = generateStatefulColor(background, positive);
  return _objectSpread({
    keySubtle: keyColors.subtle,
    keyAccent: keyColors.accent,
    keyFocus: keyColors.focus,
    keyInteractive: keyColors.interactive,
    keyPressed: keyColors.pressed,
    keyText: keyColors.text,
    keyBorder: keyColors.border,
    criticalSubtle: criticalColors.subtle,
    criticalAccent: criticalColors.accent,
    criticalFocus: criticalColors.focus,
    criticalInteractive: criticalColors.interactive,
    criticalPressed: criticalColors.pressed,
    criticalText: criticalColors.text,
    criticalBorder: criticalColors.border,
    neutralSubtle: neutralColors.subtle,
    neutralAccent: neutralColors.accent,
    neutralFocus: neutralColors.focus,
    neutralInteractive: neutralColors.interactive,
    neutralPressed: neutralColors.pressed,
    neutralText: neutralColors.text,
    neutralBorder: neutralColors.border,
    positiveSubtle: positiveColors.subtle,
    positiveAccent: positiveColors.accent,
    positiveFocus: positiveColors.focus,
    positiveInteractive: positiveColors.interactive,
    positivePressed: positiveColors.pressed,
    positiveText: positiveColors.text,
    positiveBorder: positiveColors.border
  }, extendedStatefulColors);
};
